BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/fragment/PointsOfInterestFragmentList.java

```diff
@@ -0,0 +1,91 @@
+package com.izor066.android.blocspot.ui.fragment;
+
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.izor066.android.blocspot.R;
+import com.izor066.android.blocspot.api.model.PointOfInterest;
+import com.izor066.android.blocspot.ui.MapPane;
+import com.izor066.android.blocspot.ui.adapter.ItemAdapter;
+
+/**
+ * A simple {@link Fragment} subclass.
+ */
+public class PointsOfInterestFragmentList extends Fragment implements ItemAdapter.OnPointOfInterestClickListener {
+
+
+    private RecyclerView recyclerView;
+    private ItemAdapter itemAdapter;
+
+   /* public interface OnPointOfInterestClickListener {
+        void onPointOfInterestClick(PointOfInterest pointOfInterest);
+    }
+
+    private final OnPointOfInterestClickListener listener;
+
+    public PointsOfInterestFragmentList(OnPointOfInterestClickListener listener) {
```

Comment:
Yes, this shouldn't be done as the constructor of the fragment is being used by the system in order to manage its lifecycle. 

Remember of how we talked about how the fragment communicates with its activity: the activity implements the fragment's interface and in the fragment we get hold of the listener by overriding `onAttach(Activity activity)` and inside it casting `this.listener = (MainActivity) activity`. You can find more information about this technique [here](http://developer.android.com/training/basics/fragments/communicating.html).